Enable "run" distro

**Is your feature request related to a problem? Please describe.**
The Operaton Run modules are not part of the reactor. Module `distro/run` is commented out in `/pom.xml`.

**Describe the solution you'd like**
- Activate `distro/run` in `/pom.xml` and resolve build issues.
- Adjust `nightly_build.yml` to build and publish this distro.

Enabling the `distro/run` module leads to a compile error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:testCompile (default-testCompile) on project operaton-bpm-run-qa-integration-tests: Compilation failure
[ERROR] /Users/karstenthoms/Development/projects/operaton/distro/run/qa/integration-tests/src/test/java/org/operaton/bpm/run/qa/ComponentAvailabilityIT.java:[119,9] Symbol nicht gefunden
[ERROR]   Symbol: Variable swaggerUIAvailable
[ERROR]   Ort: Klasse org.operaton.bpm.run.qa.ComponentAvailabilityIT
```
![Bildschirmfoto 2024-12-05 um 12 42 40](https://github.com/user-attachments/assets/cb2e2d27-3a0a-4ce4-8884-2167b4cb3fc6)

--- a/.github/workflows/nighly-build.yml
+++ b/.github/workflows/nighly-build.yml
@@ -41,7 +41,7 @@ jobs:
             touch "$dir/target/executeJacoco"
           fi
           done
-          ./mvnw -Pdistro,distro-tomcat,distro-wildfly,distro-webjar,distro-starter,distro-serverless,h2-in-memory verify
+          ./mvnw -Pdistro,distro-tomcat,distro-wildfly,distro-webjar,distro-starter,distro-serverless,distro-run,h2-in-memory verify
           ./mvnw --non-recursive org.jacoco:jacoco-maven-plugin:report-aggregate
       - name: Publish Test Report
         if: always()
@@ -120,6 +120,14 @@ jobs:
         with:
           name: Operaton (Tomcat ${{ env.TOMCAT_VERSION }} Bundle)
           path: distro/tomcat/distro/target/operaton-bpm-tomcat-${{ env.PROJECT_VERSION }}.tar.gz
+          if-no-files-found: error
+          retention-days: 30
+      - name: Upload distro Run
+        id: upload-distro-run
+        uses: actions/upload-artifact@v4
+        with:
+          name: Operaton Run
+          path: distro/run/distro/target/operaton-bpm-run-${{ env.PROJECT_VERSION }}.tar.gz
           if-no-files-found: error
           retention-days: 30
       - name: Upload distro Wildfly

--- a/pom.xml
+++ b/pom.xml
@@ -308,17 +308,13 @@
           any modules that have a dependency to the webapps. See also the referenced POMs,
           they usually define which modules are covered by the distro profile -->
         <module>spring-boot-starter</module>
-
         <module>quarkus-extension</module>
-
         <module>qa</module>
-
         <module>javaee/ejb-service</module>
         <module>javaee/ejb-client</module>
         <module>javaee/ejb-client-jakarta</module>
         <module>javaee/jobexecutor-ra</module>
         <module>javaee/jobexecutor-rar</module>
-
         <module>model-api</module>
         <module>engine-dmn</module>
         <module>connect</module>
@@ -326,32 +322,23 @@
         <module>spin</module>
         <module>commons</module>
         <module>freemarker-template-engine</module>
-
         <module>test-utils/testcontainers</module>
         <module>test-utils/junit5-extension</module>
         <module>test-utils/junit5-extension-dmn</module>
         <module>test-utils/assert</module>
         <module>test-utils/archunit</module>
-
         <module>engine</module>
         <module>engine-cdi</module>
         <module>engine-spring</module>
         <module>engine-rest</module>
-
         <module>webapps</module>
-
         <module>examples</module>
-
         <module>engine-plugins</module>
-
         <module>distro/license-book</module>
         <module>distro/jboss</module>
         <module>distro/tomcat</module>
         <module>distro/sql-script</module>
-        <!-- module>distro/run</module -->
-
         <module>clients/java</module>
-
         <module>internal-dependencies</module>
         <module>database</module>
         <module>parent</module>
@@ -425,7 +412,7 @@
         <activeByDefault>true</activeByDefault>
       </activation>
       <modules>
-        <!-- module>distro/run</module -->
+        <module>distro/run</module>
       </modules>
     </profile>
 

